File: java-automation/src/test/java/com/myecommerce/automation/dsl/protocols/SavedProtocol.java
package com.myecommerce.automation.dsl.protocols;

import com.myecommerce.automation.dsl.domain.SavedState;

public interface SavedProtocol extends CatalogueProtocol {

    void viewSavedItems();
    void ensureFirstProductIsSaved();
    void toggleSaveStateOfFirstProduct();
    void viewWishlist();

    SavedState getSavedState();
}

File: kotlin-automation/src/test/kotlin/com/myecommerce/automation/dsl/protocols/SavedProtocol.kt
package com.myecommerce.automation.dsl.protocols

import com.myecommerce.automation.dsl.domain.SavedState

interface SavedProtocol : CatalogueProtocol {
    fun viewSavedItems()
    fun ensureFirstProductIsSaved()
    fun toggleSaveStateOfFirstProduct()
    fun viewWishlist()

    fun getSavedState(): SavedState
}
